### DISCUSSION TITLE
Why api response for inexistent route return html page with status code 200

### DISCUSSION THREAD

**Discussion by max5432112345** [nuxt/nuxt#17029] [Questions] [answered]
### Environment

Linux

### Reproduction

Why api response for inexistent route return html page with status code 200

### Describe the bug

How to return json with proper status code 404/500 and other?

**Comment by max5432112345** [comment]
to use [Catch-all Route](https://v3.nuxtjs.org/guide/features/server-routes/#catch-all-route) with createError?

**Comment by danielroe** [comment, marked as answer]
If you only have an `app.vue` then this is used to render all routes that don't otherwise have a matcher. Likewise, if you have a catchall route `[...slug].vue`. In such cases if you wish to manually throw a 404 you may do so: https://v3.nuxtjs.org/guide/features/error-handling/#example-1

**Comment by max5432112345** [comment]
```
And 
try {
  $fetch()
catch (error) {
  console.log(error) and error is string not object and can't get code, need to parse string for 404 text
}
```
